.github/workflows/simulation.yml

```
@@ -430,14 +430,32 @@ jobs:
       uses: codecov/codecov-action@v1
 
   thread-1-2:
+    name: thread-1-2-${{ matrix.c_compiler }}-${{ matrix.arch_flags }}
     runs-on: ubuntu-18.04
     env:
-      CFLAGS: -m32
-      CXXFLAGS: -m32
-      LDFLAGS: -m32
+      CFLAGS: -${{ matrix.arch_flags }}
+      CXXFLAGS: -${{ matrix.arch_flags }}
+      LDFLAGS: -${{ matrix.arch_flags }}
       COVERAGE: 1
       THREAD_VERSION: 1.2
       VIRTUAL_TIME: 1
+      CC: ${{ matrix.c_compiler }}
+      CXX: ${{ matrix.cpp_compiler }}
+    strategy:
+      matrix:
```

Does the following syntax work?
```suggestion
      matrix:
        compiler: [{c: 'gcc', cxx: 'g++'}, { c: 'clang', cxx: 'clang++'}],
        arch: ['m32', 'm64']
```